### PR TITLE
Provide actionable guidance related to base context hash value.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #1177 by providing actionable guidance related to base context hash value. The text applies a suggestion by @jyasskin to treat the context URLs as "already resolved" with specific hash values. It also notes that implementations are expected to throw errors when hash values differ from the expected value w/o saying that they MUST throw errors (in cases where hash values change all the time, like schema.org).

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1249.html" title="Last updated on Aug 27, 2023, 11:01 PM UTC (aec155a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1249/43ac552...aec155a.html" title="Last updated on Aug 27, 2023, 11:01 PM UTC (aec155a)">Diff</a>

### CODE REVIEW

**Review by dlongley — APPROVED** [review]

**Review by Wind4Greg — APPROVED** [review]

**Review by iherman — APPROVED** [review]
I cannot directly comment on this using the github tool, hence the separate commen. The text says:

> The contents of the credential itself, which is secured in
<a>verifiable credentials</a> and <a>verifiable presentations</a> by using
mechanisms such as [[VC-JOSE-COSE]] and [[VC-DATA-INTEGRITY]].

I would prefer to use "or" instead of "and" between jose-cose and di. Just to avoid misunderstandings (easy for non-native) that the application is required to use _both_.

File: index.html [hunk 1]
```diff
@@ -5452,24 +5452,25 @@ <h3>Base Context</h3>
...
-        </p>
-        <p class="issue" data-number="1177">
-The Working Group is currently discussing what a processor should do if a hash
-value differs from one that is listed in the specification.
-        </p>
-More details regarding this hash encoding method may be found in the <a
+Implementations MUST treat the base context value, located at
+<code>https://www.w3.org/ns/credentials/v2</code>, as already resolved where the
+content of the resource matches the following SHA-384 digest of the value
+computed and encoded according to the [[SRI]] definition of `digest`:
```
[the remark this answers is not in the document]
Comment (msporny) [reply index.html:5458]: Hrm... "if" implies that they should try and resolve it rather than never try to resolve it and always use a local copy to resolve it. That's what we're trying to say here -- "Just use a local copy and that local copy MUST match the provided hash". 

I'll think about this some more and try to provide alternative language that's more straight forward.

File: index.html [hunk 2]
```diff
@@ -5495,9 +5496,9 @@ <h3>Base Context</h3>
 credential, such as resources that are linked to via URLs, are not
 cryptographically protected by default. It is considered a best practice to
 ensure that the same sorts of protections are provided for any URL that is
-critical to the security of the credential through the use of permanently cached
-files and/or cryptographic hashes. See the <a
-data-cite="vc-imp-guide/#content-integrity">Content Integrity</a>
+critical to the security of the <a>verifiable credential</a> through the use of
+permanently cached files and/or cryptographic hashes. See the
```
Comment (decentralgabe) [inline index.html:5500]: are these files intended to be immutable?
if not, is there a TTL that informs caching behavior?

**Review by OR13 — COMMENTED** [review]
@msporny @iherman can we align this with the thread on the http headers for serving the file?.. perhaps a warning that the hash will change if there are errata, and therefore the cache time header is not set to indefinitely?

File: index.html (hunk 2 above)
Comment (msporny) [reply index.html:5500]: Once the VCWG takes the spec to REC, the files are intended to be immutable (modulo the errata statement @OR13 mentions below https://github.com/w3c/vc-data-model/pull/1249#pullrequestreview-1587450604).

**Comment by msporny** [comment]
> @msporny @iherman can we align this with the thread on the http headers for serving the file?.. perhaps a warning that the hash will change if there are errata, and therefore the cache time header is not set to indefinitely?

Sure, I can put some sort of note to that effect in the section as well.

**Comment by iherman** [comment]
> > @msporny @iherman can we align this with the thread on the http headers for serving the file?.. perhaps a warning that the hash will change if there are errata, and therefore the cache time header is not set to indefinitely?
> 
> Sure, I can put some sort of note to that effect in the section as well.

+1 for me as well.

(B.t.w., I still expect some reviews on the changes of the .htacess file in #1239. For the time being, no change has been made on the access file in that respect...

File: index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply index.html:5458]: Should `already resolved ` be `already dereferenced` or perhaps `already retrieved`? I ask because elsewhere, "resolution" is being used to describe conversion from "relative URL" to what is often (apparently incorrectly) called "absolute URL".

[the remark this answers is not in the document]
Comment (msporny) [reply index.html:5458]: Yeah, perhaps "already retrieved" is the best word we could use for people that don't know the differences between "resolved" and "dereferenced"... to this day, I still have to look up the proper terminology when authoring specs.

[the remark this answers is not in the document]
Comment (msporny) [reply index.html:5458]: Hrm, no "if" isn't the right word. The content at that resource will be the hash value, full stop.

[the remark this answers is not in the document]
Comment (msporny) [reply index.html]: Fixed in 31bf381baca615410f10cf1830cdad423be96180.

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.
